README.md

```diff
@@ -1 +1,59 @@
-# MIRAD
+# MIRAD: A Method for Interpretable Ransomware Attack Detection
+
+In this repository we open-source a method for training interpretable models,
+originally developed for ransomware attack detection.
+
+Key features:
+- model and prediction interpretability
+- concise implementation based on `numpy` and `pytorch`
+- efficiency: training can be executed on a CPU, evaluation is fast
+- extensibility: model structure allows easy enforcement of task-specific requirements by additional regularization
+
+We also include everything necessary to reproduce experiments we conducted to publish our results.
+
+## How it works
+
+All the implementation is in [lib/embedding_sum.py](lib/embedding_sum.py) and consists of three classes:
+ - Digitizer
+ - EmbeddingSumModule
+ - EmbeddingSumClassifier
+
+The method can be summarized as follows:
+
+1. For each feature, the `Digitizer` defines quantile-based bins and encodes feature values as bin ordinals,
+2. The `EmbeddingSumModule` defines a parameter for each bin of each feature and implements evaluation as sum of the
+   parameters corresponding to bin ordinals in the input vector,
```

Comment:
The word parameter is very vague. Is there another term that doesn't go against the literature that you could use? Something more descriptive.

Reply:
now: trainable parameter